### ISSUE TITLE
plotly and Mathjax

### ISSUE THREAD

**Issue by benjaminleroy** [r-lib/pkgdown#1338]
Hi there,

I've found that plotly and mathjax don't work together when they are included in a vignette.

Below you'll find a basic example (that seems to fit your contributing guidelines.

Thanks

-------

<pre>
library(usethis)
library(pkgdown)
library(plotly)

tmp <- file.path(tempdir(), "errordemo/")
usethis::create_package(tmp, open = FALSE)

dir.create(paste0(tmp, "vignettes/"))
write(x = "
---
title: \"error_demo\"
output: rmarkdown::html_vignette
vignette: >
  %\\VignetteIndexEntry{error_demo}
  %\\VignetteEngine{knitr::rmarkdown}
  %\\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = \"#>\"
)
```

```{r setup}
library(errordemo)
```

\\[ 
  Y = X\\beta + \\epsilon
\\] 

```{r message = F, warning = F}
library(plotly)

fig <- plot_ly() %>% 
  add_trace(x = c(1,0,0),
            y = c(0,1,0),
            z = c(0,0,1),
            facecolor = \"grey\",
            opacity = .5,
            type = \"mesh3d\") 

fig
```

\\[ 
\\hat{\\beta} = (X^TX)^{-1}X^Ty
\\]", 
      file = paste0(tmp,"vignettes/error_demo.Rmd"))

pkgdown::build_site(tmp, new_process = FALSE, preview = TRUE)

</pre>

I get the following back (in the article):

<img width="777" alt="Screen Shot 2020-05-29 at 11 37 43 AM" src="https://user-images.githubusercontent.com/8593016/83280302-70f47980-a1a4-11ea-9c15-729049e7f208.png">

**Comment by jayhesselberth** [comment]
Please see the vignette for the correct Mathjax syntax:

https://github.com/r-lib/pkgdown/blob/d68907726c41b8f6595dd440e1a573db08d0ae72/vignettes/test/rendering.Rmd#L104-L108

**Comment by benjaminleroy** [comment]
Hi @jayhesselberth I've updated to use `$$` and the error presists.

**Comment by jayhesselberth** [comment]
This appears to be an interaction with the Mathjax library that pkgdown loads and the Mathjax library that is included in plotly. Perhaps there is a way to suppress use of mathjax by `plot_ly()`? If it's included automatically, it won't be an easy fix.

**Comment by benjaminleroy** [comment]
TL:DR: did some brief exploring, `plotly`'s recommended fix still fails with `pkgdown`, both options don't have a problem with `rmarkdown::render`, will do more exploring.


Thanks for the suggestion. `plotly` does provide a recommendation like you suggested that I used (https://rdrr.io/cran/plotly/f/inst/examples/rmd/MathJax/index.Rmd) - which just wraps around the `plotly` visual with `widgetframe::frameableWidget`,  sadly it doesn't appear to work with `pkgdown` html rendering.

Below is updated code the still fails to render:

<pre>
library(usethis)
library(pkgdown)
library(plotly)
library(widgetframe)

tmp <- file.path(tempdir(), "errordemo/")
usethis::create_package(tmp, open = FALSE)

dir.create(paste0(tmp, "vignettes/"))
write(x = "
---
title: \"error_demo\"
output: rmarkdown::html_vignette
vignette: >
  %\\VignetteIndexEntry{error_demo}
  %\\VignetteEngine{knitr::rmarkdown}
  %\\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = \"#>\"
)
```

```{r setup}
library(errordemo)
```

$$ 
  Y = X\\beta + \\epsilon
$$

```{r message = F, warning = F}
library(plotly)

fig <- plot_ly() %>% 
  add_trace(x = c(1,0,0),
            y = c(0,1,0),
            z = c(0,0,1),
            facecolor = \"grey\",
            opacity = .5,
            type = \"mesh3d\") 

widgetframe::frameableWidget(fig)
```

$$
\\hat{\\beta} = (X^TX)^{-1}X^Ty
$$", 
file = paste0(tmp,"vignettes/error_demo.Rmd"))

pkgdown::build_site(tmp, new_process = FALSE, preview = TRUE)
</pre>

What's interesting to me (and the reason I created this issue) is because the conflict between Matjax and `plotly` doesn't appear when I use `rmarkdown::::render` to html (a demo of such things can be seen in the following code (note I removed a call to the "errordemo" package as it doesn't exist globally):

<pre>
library(usethis)
library(pkgdown)
library(plotly)

tmp <- file.path(tempdir(), "errordemo/")
usethis::create_package(tmp, open = FALSE)

dir.create(paste0(tmp, "vignettes/"))
write(x = "
---
title: \"error_demo\"
output: rmarkdown::html_vignette
vignette: >
  %\\VignetteIndexEntry{error_demo}
  %\\VignetteEngine{knitr::rmarkdown}
  %\\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = \"#>\"
)
```

$$ 
  Y = X\\beta + \\epsilon
$$

```{r message = F, warning = F}
library(plotly)

fig <- plot_ly() %>% 
  add_trace(x = c(1,0,0),
            y = c(0,1,0),
            z = c(0,0,1),
            facecolor = \"grey\",
            opacity = .5,
            type = \"mesh3d\") 

fig
```

$$
\\hat{\\beta} = (X^TX)^{-1}X^Ty
$$", 
file = paste0(tmp,"vignettes/error_demo.Rmd"))

rmarkdown::render(paste0(tmp,"vignettes/error_demo.Rmd"), 
                                 output_file = paste0(tmp,"vignettes/error_demo.html"))
</pre>

Either way, I'll explore this more on my own. 

(also - should this issue be closed?)

**Comment by benjaminleroy** [comment]
**Just in case someone needs the solution** - below demonstrates a solution using `widgetframe` and saving the files for pkgdown through the `man/figures` folder (which get's saved into [pkgdown's website](https://github.com/r-lib/pkgdown/issues/280)). widgetframe also makes this harder due to this [issue](https://github.com/ramnathv/htmlwidgets/issues/296), so we use the recommended hack.

<pre>
library(usethis)
library(pkgdown)
library(plotly)
library(widgetframe)

tmp <- file.path(tempdir(), "errordemo/")
usethis::create_package(tmp, open = FALSE)

dir.create(paste0(tmp, "vignettes/"))
dir.create(paste0(tmp, "man/"))
dir.create(paste0(tmp, "man/figures/"))

write(x = "---
title: \"error_demo\"
output: rmarkdown::html_vignette
vignette: >
  %\\VignetteIndexEntry{error_demo}
  %\\VignetteEngine{knitr::rmarkdown}
  %\\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = \"#>\"
)
```

```{r setup}
library(errordemo)
```

$$ 
  Y = X\\beta + \\epsilon
$$

```{r message = F, warning = F, eval = F}
library(plotly)

fig <- plot_ly() %>% 
  add_trace(x = c(1,0,0),
            y = c(0,1,0),
            z = c(0,0,1),
            facecolor = \"grey\",
            opacity = .5,
            type = \"mesh3d\") 
```

```{r echo = F}
### the following line of code was run once:
# htmlwidgets::saveWidget(fig, file = paste0(tmp, \"man/figures/plotly_saved.html\"),
#                        selfcontained=TRUE, libdir = paste0(tmp, \"man/figures/not_a_real_directory_name\"))

htmltools::tags$iframe(
  src = \"../reference/figures/plotly_saved.html\",
  scrolling = \"no\", 
  seamless = \"seamless\",
  frameBorder = \"0\",
  width=\"100%\", 
  height=\"400\"
)
```

$$
\\hat{\\beta} = (X^TX)^{-1}X^Ty
$$", 
file = paste0(tmp,"vignettes/error_demo.Rmd"))



pkgdown::build_site(tmp, new_process = FALSE, preview = TRUE)
</pre>
